### ISSUE TITLE
Incorrect inference of default literal expression

### ISSUE THREAD

**Issue by smitpatel** [dotnet/roslyn#22505]
I am trying out C# 7.1 feature default literal expression.
```
    class Program
    {
        static void Main(string[] args)
        {
            var a = Expression.Constant(default(int), typeof(int));
            Console.WriteLine(a.Value);
            Console.WriteLine("Hello World!");
        }
    }
```
Generates output
```
PM> dotnet run
0
Hello World!
```
But VS shows me code fix that default expression can be simplified.
![image](https://user-images.githubusercontent.com/1528107/31153072-e8ceb468-a853-11e7-9638-e8a288420a7b.png)
When I get VS to simplify the code
```
        static void Main(string[] args)
        {
            var a = Expression.Constant(default, typeof(int));
            Console.WriteLine(a.Value);
            Console.WriteLine("Hello World!");
        }
```
The code throws exception
```
PM> dotnet run
dotnet : 
At line:1 char:1
+ dotnet run
+ ~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Unhandled Exception:
 
System.ArgumentException: Argument types do not match
   at System.Linq.Expressions.Expression.Constant(Object value, Type type)
   at DefaultLiteral.Program.Main(String[] args) in C:\users\smpatel\Source\Repos\DefaultLiteral\DefaultLiteral\Program.cs:line 10
```

**Comment by jcouv** [comment]
FYI @sharwell @CyrusNajmabadi

**Comment by sharwell** [comment]
:memo: Probable duplicate of #22184.

**Comment by CyrusNajmabadi** [comment]
Indeed.  I thought this was fixed with the change to ensure that the type of the ```default``` literal after hte change was the same as the type of the ```default()``` expression before the change.  Was that fix not checked in?

**Comment by jcouv** [comment]
I verified that the fix isn't offered in 15.4.1.
I'll go ahead and close the issue. Thanks
